reverse.py

```diff
@@ -0,0 +1,40 @@
+from string import punctuation, digits
+
+def reversed_func(text):
+    """
+    The function reverses the letters in all words of input text.
+    The order of words preserves.
+    All non-letter symbols/numbers stays on the same places.
+    """
+    PRESERVED_CHARS = punctuation + digits
+    reversed_text = []
+
+    for word in text.split():
+        char_list = []
+        reversed_word = ''
+        for char in word:
+            if char not in PRESERVED_CHARS:
+                char_list.append(char)
+
+        for char in word:
+            if char in PRESERVED_CHARS:
+                reversed_word += char
```

Comment:
Appending character to the string `reversed_word ` is a compute-complex action (technically). String is an immutable data type so by appending new char to the string, you technically force python to re-create variable in memory, and delete the old one. Please fix and check out mutable/immutable data types in python